### PR TITLE
feat: allow to create item with thumbnail

### PR DESCRIPTION
**Pull request by spaenleh**

### CODE REVIEW

File: src/services/file/service.ts [hunk 1]
```diff
@@ -214,8 +213,8 @@ class FileService {
     replyUrl?: boolean;
   }) {
     if (replyUrl) {
-      const replyUrlExpiration = S3_PRESIGNED_EXPIRATION;
-      reply.header('Cache-Control', `max-age=${replyUrlExpiration}`);
+      // const replyUrlExpiration = S3_PRESIGNED_EXPIRATION;
+      // reply.header('Cache-Control', `max-age=${replyUrlExpiration}`);
```
Comment (pyphilia) [inline src/services/file/service.ts:217]: Remove.

File: src/services/item/test/index.test.ts [hunk 2]
```diff
@@ -485,6 +522,44 @@ describe('Item routes tests', () => {
...
+      payload.append('file', imageStream);
+      const response = await app.inject({
+        method: HttpMethod.Post,
+        url: `/items/with-thumbnail`,
+        payload,
+        headers: payload.getHeaders(),
+      });
+
+      const newItem = response.json();
+      console.log(newItem);
```
Comment (pyphilia) [inline src/services/item/test/index.test.ts:546]: Remove.

File: src/services/item/utils.ts [hunk 3]
```diff
@@ -91,3 +104,93 @@ export const readPdfContent = async (source: string | URL) => {
...
+  // name should be present
+  if (!name) {
+    throw new Error('missing required name');
+  }
+  const maybeType = getFieldFromMultipartForm(formData.fields, 'type');
+  // type should be present and should be one of the available types
+  if (!maybeType || !(Object.values(ItemType) as string[]).includes(maybeType)) {
+    throw new Error('missing type or invlid type provided');
+  }
+  // here we cast as we have checked previously that it was valid
```
Comment (pyphilia) [inline src/services/item/utils.ts:158]: After discussion, I think it's better to get the full body and validate with a single schema (probably a schema we already have).

File: src/services/item/validation.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import Ajv from 'ajv';
+
+import { ItemGeolocation } from '@graasp/sdk';
+
+import { Item } from './entities/Item';
+import { geolocation, settings } from './fluent-schema';
+
+/**
+ * Decalre compiled validators to be used in manual validation
```
Comment (pyphilia) [inline src/services/item/validation.ts:9]:
```suggestion
 * Declare compiled validators to be used in manual validation
```

File: src/services/item/validation.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+ */
+const ajv = new Ajv({ allErrors: true });
+export const validateSettings = ajv.compile<Item['settings']>(settings.valueOf());
+// const validateExtra = ajv.compile(extra.valueOf());
```
Comment (pyphilia) [inline src/services/item/validation.ts:13]: Remove? Needed?

**Review by pyphilia — APPROVED** [review]
You can move some comments to other issues. Thanks for the PR! 🦆

File: src/services/item/utils.ts (hunk 3 above)
Comment (spaenleh) [reply src/services/item/utils.ts:158]: Yes, I created #1011

**Pull request merged**